ambry-router/src/test/java/com/github/ambry/router/DeleteManagerTest.java

```diff
@@ -130,20 +131,21 @@ public void init() throws Exception {
             BlobId.BlobDataType.DATACHUNK);
     blobIdString = blobId.getID();
     deleteManager =
-        new DeleteManager(clusterMap, new ResponseHandler(clusterMap), accountService, notificationSystem,
-            routerConfig, routerMetrics, new RouterCallback(null, new ArrayList<>()), mockTime);
+        new DeleteManager(clusterMap, new ResponseHandler(clusterMap), accountService, notificationSystem, routerConfig,
+            routerMetrics, new RouterCallback(null, new ArrayList<>()), mockTime);
     MockNetworkClientFactory networkClientFactory =
         new MockNetworkClientFactory(vProps, mockSelectorState, MAX_PORTS_PLAIN_TEXT, MAX_PORTS_SSL,
             CHECKOUT_TIMEOUT_MS, serverLayout, mockTime);
     networkClient = networkClientFactory.getNetworkClient();
+    operationCountBeforeTest = router.getOperationsCount();
```

Comment:
how can we know op count before test ? Also shouldn't it be deterministically zero since we just started the test ?

Reply:
the issue with operation count is that it's a static and yet mutable variable. So router.getOperationsCount() would just return this value. Since it's static, different test cases would interfere each other. If there is one test that increases operation count but didn't wait for it to finish, then operation count would not be zero any more for other tests. With this change, we would compare operation counts before and after tests and to make sure the tests that care about the operation counts, have the proper way to verify all the operations are done.

Reply:
I see, please put this in the commit description so when look at the history we know the reason for this change.